arch/arm/core/aarch32/cortex_m/CMakeLists.txt

```diff
@@ -14,6 +14,7 @@ zephyr_library_sources(
   )
 
 zephyr_library_sources_ifdef(CONFIG_DEBUG_COREDUMP coredump.c)
+zephyr_library_sources_ifdef(CONFIG_CORTEX_M_DWT timing.c)
```

Comment:
I do not think this is correct, should depend on CONFIG_TIMING_FUNCTIONS instead.

Reply:
I wanted to
-  have a DWT-specific Kconfig switch to enable compiling the timing.c
- have a way to prioritize DWT's timing.c against SoC's timing.c (see later)